src/Fields/broadcasting_abstract_fields.jl

```diff
@@ -0,0 +1,105 @@
+
+# Precedence rule
+Base.Broadcast.BroadcastStyle(::FieldBroadcastStyle, ::DefaultArrayStyle{N}) where N = FieldBroadcastStyle()
+
+# For use in Base.copy when broadcasting with numbers and arrays (useful for comparisons like f::AbstractField .== 0)
+Base.similar(bc::Broadcasted{FieldBroadcastStyle}, ::Type{ElType}) where ElType = similar(Array{ElType}, axes(bc))
+
+# Bypass style combining for in-place broadcasting with arrays / scalars to use built-in broadcasting machinery
+@inline Base.Broadcast.materialize!(dest::AbstractField, bc::Broadcasted{<:DefaultArrayStyle}) =
+    Base.Broadcast.materialize!(DefaultArrayStyle{3}(), dest, bc)
+
+#####
+##### Kernels
+#####
+
+@kernel function broadcast_xyz!(dest, bc)
+    i, j, k = @index(Global, NTuple)
+    @inbounds dest[i, j, k] = bc[i, j, k]
+end
+
+@kernel function broadcast_xy!(dest, bc)
+    i, j = @index(Global, NTuple)
+    @inbounds dest[i, j, 1] = bc[i, j, 1]
+end
+
+@kernel function broadcast_xz!(dest, bc)
+    i, k = @index(Global, NTuple)
+    @inbounds dest[i, 1, k] = bc[i, 1, k]
+end
+
+@kernel function broadcast_yz!(dest, bc)
+    j, k = @index(Global, NTuple)
+    @inbounds dest[1, j, k] = bc[1, j, k]
+end
+
+# Three-dimensional general case
+
+launch_configuration(::AbstractField) = :xyz
+broadcast_kernel(::AbstractField) = broadcast_xyz!
+
+# Two dimensional reduced field
+
+const Loc = Union{Center, Face}
+
+broadcast_kernel(::AbstractReducedField{Nothing, <:Loc, <:Loc}) = broadcast_yz!
+broadcast_kernel(::AbstractReducedField{<:Loc, Nothing, <:Loc}) = broadcast_xz!
+broadcast_kernel(::AbstractReducedField{<:Loc, <:Loc, Nothing}) = broadcast_xy!
+
+launch_configuration(::AbstractReducedField{Nothing, <:Loc, <:Loc}) = :yz
+launch_configuration(::AbstractReducedField{<:Loc, Nothing, <:Loc}) = :xz
+launch_configuration(::AbstractReducedField{<:Loc, <:Loc, Nothing}) = :xy
+
+broadcasted_to_abstract_operation(loc, grid, a) = a
+
+#####
+##### Hopefully we don't need to interpolate, but if we do...
+#####
+
+needs_interpolation(Xa, Xb) = false
+needs_interpolation(::Type{Center}, ::Type{Face}) = true
+needs_interpolation(::Type{Face}, ::Type{Center}) = true
+
+needs_interpolation(La::Tuple, Lb::Tuple) = any(needs_interpolation.(La, Lb))
+needs_interpolation(La::Tuple, b::AbstractField) = needs_interpolation(La, location(b))
+needs_interpolation(La::Tuple, ::Number) = false
+needs_interpolation(La::Tuple, ::AbstractArray) = false
+needs_interpolation(La::Tuple, bc::Broadcasted) = any(needs_interpolation(La, b) for b in bc.args)
+
+# Broadcasting with interpolation breaks Base's default rules for ABroBbstractOperations 
+@inline Base.Broadcast.materialize!(::Base.Broadcast.BroadcastStyle,
+                                    dest::AbstractField,
+                                    bc::Broadcasted{<:FieldBroadcastStyle}) = copyto!(dest, convert(Broadcasted{Nothing}, bc))
+
+# Non-interpolated broadcasting to Field
```

Comment:
What currently happens if you try to broadcast two fields at different locations? I'm guessing it should error with a message saying it's currently unsupported?

Reply:
This works! There's a test for it. The comment is wrong. I'll change it.